OIDC configuration (#1375)
spectrocloud/librarium: content/docs/06-integrations/00-spectro-k8s-dashboard.md

# Spectro Kubernetes Dashboard

> Spectro Kubernetes Dashboard is a web-based UI for Kubernetes clusters that auto-enables the Kubernetes Dashboard using secure ports and conveniently includes [Spectro Proxy](/integrations/frp). 

Reviewer: ```suggestion
Spectro Kubernetes Dashboard is a web-based UI for Kubernetes clusters that auto-enables the Kubernetes Dashboard using secure ports and conveniently includes the [Spectro Proxy](/integrations/frp) pack. 
```
Minor clarification to help those new to the platform.

## Prerequisites

- Outbound internet connectivity for port 443 is allowed so that you and your applications can connect with the [Spectro Proxy](/integrations/frp) reverse proxy.

> - Users or groups must be mapped to a Kubernetes RBAC role, either a *Role* or a *ClusterRole*. You can create a custom role through a manifest and use Palette's roleBinding feature to associate users or groups with the role. Refer to the [Create a Role Binding](/clusters/cluster-management/cluster-rbac#createrolebindings) guide to learn more. 

Reviewer: Good 😄

| `certRenewal` | Self-signed certificate renewal in hours | 720h (30d) |
| `enableInsecureLogin` | RBAC ensures secure login. | `false` |
| `serviceType` | The service type for the dashboard. Supported values are `ClusterIP`, `LoadBalancer`, and `NodePort`. | `ClusterIP` |
| `skipLogin` | Enables or disables the skip login option on the Spectro Kubernetes Dashboard. | `false` |
| `enableInsecureLogin` | Enables non-Secure Sockets Layer (SSL) login. Dashboard login is always restricted to HTTP(S) + localhost or HTTPS and external domain. | `false` |
| `ingress.enabled` | Ingress configuration to access the `ClusterIP`, `LoadBalancer`, or `NodePort`. | `false` |

## Usage

To use the Spectro Kubernetes Dashboard pack, you have to add it to your cluster profile. Spectro Kubernetes Dashboard supports several public cloud and data center cloud environments. To learn more, refer to [Clusters](/clusters).

Use the following information to find the Kubernetes Dashboard pack.
- **Pack Type**: Monitoring
- **Registry**: Public Repo
- **Pack Name**: Spectro Kubernetes Dashboard
- **Pack Version**: 2.7.0 or higher

Spectro Kubernetes Dashboard has the following Access options.

<br /> 

> - **Proxied**: This option provides private access to your cluster when the cluster is in a private cloud. This option requires the Spectro Proxy pack, which is integrated with the Spectro Kubernetes Dashboard. The latest version of the Spectro Proxy pack is automatically installed when you create the cluster. To learn more, check out the [Spectro Proxy](/integrations/frp) guide.

Reviewer 2: ```suggestion
- **Proxied**:  This option is useful for access to the Kubernetes Dashboard from anywhere. It will leverage the Spectro Proxy pack for this, which gets automatically installed if not present. The Service resource for the Kubernetes Dashboard will be configured as ClusterIP and is only accessible through the proxy. To learn more, check out the [Spectro Proxy](/integrations/frp) guide.
```

Author: I committed this, but it doesn't seem to indicate that.

Author: I added a slightly worded version of this to the doc. Thank you @kreeuwijk

> - **Direct**: This option provides direct access to your cluster when the cluster is in a public cloud.

Reviewer 2: ```suggestion
- **Direct**: This option is meant for private configuration, where a proxy is not implemented or not desired. The Service resource for the Kubernetes Dashboard will be configured as LoadBalancer and gets directly accessed by the end user. This option requires that the user is on a network that can reach the IP address given to the LoadBalancer service.
```

Reviewer: Much appreciated for the additional context @kreeuwijk 🙏🏻

Author: Yes, thank you @kreeuwijk !

Author: Committed.

Author: I added a slightly worded version of this to the doc. Thank you @kreeuwijk
